debug_toolbar/store.py

```diff
@@ -140,5 +142,100 @@ def panels(cls, request_id: str) -> Any:
             yield panel, deserialize(data)
 
 
+class DatabaseStore(BaseStore):
+    @classmethod
+    def _cleanup_old_entries(cls):
+        """
+        Enforce the cache size limit - keeping only the most recently used entries
+        up to RESULTS_CACHE_SIZE.
+        """
+        # Get the cache size limit from settings
+        cache_size = dt_settings.get_config()["RESULTS_CACHE_SIZE"]
+
+        # Determine which entries to keep (the most recent ones up to cache_size)
+        keep_ids = list(
+            DebugToolbarEntry.objects.order_by("-created_at")[:cache_size].values_list(
+                "request_id", flat=True
+            )
+        )
+
+        # Delete all entries not in the keep list
+        if keep_ids:
+            DebugToolbarEntry.objects.exclude(request_id__in=keep_ids).delete()
```

Comment:
Unimportant/nitpicky: I think blindly running the `delete` query would be fine; you could remove the `list()` call above then. I haven't tested if that works, and if the `list()` call is necessary for some reason then I'm fine with this.